### PR TITLE
refactor: Annotated is now a proper class

### PR DESCRIPTION
**Pull request by prestwich**
Make Annotated a class so we can have a bunch of nice getters and stuff :)

### CODE REVIEW

File: typescript/optics-provider/src/optics/events/opticsEvents.ts [hunk 1]
```diff
@@ -34,9 +36,70 @@ export type ProcessEvent = TypedEvent<ProcessTypes & ProcessArgs>;
...
+export class Annotated<U extends Result, T extends TypedEvent<U>> {
+  readonly domain: number;
+  readonly eventName?: string;
+  readonly event: T;
+  readonly receipt: TransactionReceipt;
+  constructor(
+    domain: number,
+    receipt: TransactionReceipt,
+    event: T,
+    callerKnowsWhatTheyAreDoing = false,
```
Comment (anna-carroll) [inline typescript/optics-provider/src/optics/events/opticsEvents.ts:48]: i don't like this but i don't want to scrap about it 🌝

**Review by anna-carroll — APPROVED** [review]
overall LGTM

File: typescript/optics-provider/src/optics/contracts/BridgeContracts.ts [hunk 2]
```diff
@@ -4,6 +4,16 @@ import { Contracts } from '../../contracts';
 
 type Address = string;
 
+interface ProxyInfo {
+  proxy: Address;
+}
+
+interface BridgeInfo {
+  id: number;
+  bridgeRouter: Address | ProxyInfo;
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/contracts/BridgeContracts.ts:13]: It's not that it won't be a proxy, it's that we'd like to allow multiple shapes of data for this input. I.e. allow someone to omit the UB and Implementation info and just provide the proxy address

File: typescript/optics-provider/src/optics/messages/OpticsMessage.ts [hunk 3]
```diff
@@ -320,7 +324,9 @@ export class OpticsMessage {
   /// May never resolve. May take hours to resolve.
   async wait(opts?: { pollTime?: number }): Promise<void> {
     const interval = opts?.pollTime ?? 5000;
-    while (true) {
+
+    // sad spider face
+    for (;;) {
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/messages/OpticsMessage.ts:329]: it's the eslint-approved way to make an unconditional loop.

File: typescript/optics-provider/src/optics/messages/OpticsMessage.ts [hunk 4]
```diff
@@ -355,12 +361,14 @@ export class OpticsMessage {
...
-
+  get bodyHash(): string {
+    return keccak256(this.body);
+  }
   get transactionHash(): string {
     return this.dispatch.event.transactionHash;
   }
 
-  get messageHash(): string {
+  get leaf(): string {
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/messages/OpticsMessage.ts:371]: standardizing terminology w/ the rust codebase and removing ambiguity with the bridge message

File: typescript/optics-provider/src/optics/messages/OpticsMessage.ts [hunk 5]
```diff
@@ -355,12 +361,14 @@ export class OpticsMessage {
   get body(): string {
     return this.message.body;
   }
-
+  get bodyHash(): string {
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/messages/OpticsMessage.ts:364]: yes, it's used for the prefill ID in bridge messages

File: typescript/optics-provider/src/optics/events/opticsEvents.ts (hunk 1 above)
Comment (prestwich) [reply typescript/optics-provider/src/optics/events/opticsEvents.ts:48]: haha i don't like it either, but it is the standard way to gate constructors. the variable name is me being annoyed about it

File: typescript/optics-provider/src/optics/messages/OpticsMessage.ts [hunk 6]
```diff
@@ -117,17 +119,19 @@ export class OpticsMessage {
               .getTransactionReceipt(log.transactionHash);
           };
 
-          const annotated = annotate(
+          const annotated = new Annotated<DispatchTypes, DispatchEvent>(
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/messages/OpticsMessage.ts:122]: we can add that as an alias in another PR

File: typescript/optics-provider/src/optics/messages/OpticsMessage.ts [hunk 7]
```diff
@@ -117,17 +119,19 @@ export class OpticsMessage {
...
+            true,
           );
-          annotated.name = 'Dispatch';
           annotated.event.blockNumber = annotated.receipt.blockNumber;
           const message = new OpticsMessage(context, annotated);
           messages.push(message);
         }
-      } catch (e) {}
+      } catch (e) {
+        continue;
```
[the remark this answers is not in the document]
Comment (prestwich) [reply typescript/optics-provider/src/optics/messages/OpticsMessage.ts:133]: eslint hates empty blocks, so I'm making the behavior explicit. will add comment on this as a cleanup item

**Pull request merged**